source/NVDAObjects/IAccessible/ia2Web.py

```diff
@@ -215,6 +219,8 @@ def findExtraOverlayClasses(obj, clsList, baseClass=Ia2Web, documentClass=None):
 	xmlRoles = obj.IA2Attributes.get("xml-roles", "").split(" ")
 	if iaRole == IAccessibleHandler.IA2_ROLE_SECTION and obj.IA2Attributes.get("tag", None) == "blockquote":
 		clsList.append(BlockQuote)
+	elif iaRole == oleacc.ROLE_SYSTEM_OUTLINE and "treegrid" in xmlRoles:
+		clsList.append(Treegrid)

```

Comment:
I don't see that this is currently necessary? Is this just done for future convenience / due to convention?

Reply:
the TreeGrid class forces the role to table  so that focus mode / object navigation reports it as a table, not treeview. Remembering that the changes in the vbufBackend only affect  the virtualBuffer (browse mode).

Reply:
Oh, sorry. Somehow I didn't note the importance of `role = controlTypes.ROLE_TABLE` in the `Treegrid` class.